js/stringProcessing/htmlParser.js

```diff
@@ -0,0 +1,71 @@
+// We use an external library, which can be found here: https://github.com/fb55/htmlparser2.
+let htmlparser = require( "htmlparser2" );
+
+// The array containing the text parts without the blocks defined in inlineTags.
+let textArray;
+
+// False when we are not in a block defined in inlineTags. True if we are.
+let inScriptBlock = false;
+
+// The blocks we filter out of the text that needs to be parsed.
+let inlineTags = [ "script", "style" ];
+
+/**
+ * Parses the text.
+ */
+let parser = new htmlparser.Parser( {
+	/**
+	 * Handles the opening tag. If the opening tag is included in the inlineTags array, set inScriptBlock to true.
+	 * If the opening tag is not included in the inlineTags array, push the tag to the textArray.
+	 *
+	 * @param {string} tagName The tag name.
+	 *
+	 * @returns {void}
+	 */
+	onopentag: function( tagName ) {
+		if ( inlineTags.includes( tagName ) ) {
+			inScriptBlock = true;
+		} else {
+			textArray.push( "<" + tagName + ">" );
```

Comment:
When fixing this, also add a unit test for this behavior. When writing unit tests try to write tests that are outside the happy path.